photon-lib/src/main/scala/com/linkedin/photon/ml/Types.scala

```diff
@@ -27,14 +27,17 @@ object Types {
   // A "sample" is a training, validation or scoring vector of feature values
   type UniqueSampleId = Long
 
-  // A CoordinateId is a fixed effect type, random effect type, or matrix factorization model type
+  // A CoordinateId corresponds to a fixed effect type, random effect type, or matrix factorization model type
   // (e.g. "per-user", "per-song" ...)
   type CoordinateId = String
 
   // RE stands for "random effect"
+  // REType is e.g. "per-song", "per-user"
   type REType = String
+  // REId is e.g. the song id, user id for which a random effect of the right type will be optimized
```

Comment:
What do you think about:
`REId is a specific instance of a REType e.g. "user123", "user456" for "per-user"`